22_23_10/script.js

```diff
@@ -0,0 +1,61 @@
+const form = document.getElementById("mainForm");
+
+class InputParent {
+  constructor(name, type) {
+    this.name = name;
+    this.type = type;
+  }
+
+  createInput(attr) {
+    this.element = document.createElement("input");
+    this.element.type = this.type;
+    this.element.setAttribute(attr, this.attrValue);
+  }
+  createLabel(text) {
+    const label = document.createElement("label");
+    label.textContent = text;
+    label.setAttribute("for", this.id);
+    label.prepend(this.element);
+    form.append(label);
+  }
+}
+
+class Input extends InputParent {
+  constructor(name, type) {
+    super(name, type);
+  }
```

Comment:
И если ждешь параметры виде объекта, то тут можно делать так:
```suggestion
  constructor(props) {
    super(props);
  }
```